### PR TITLE
Perf tests for overview page and varying namespaces

### PR DESCRIPTION
**Pull request by nrfox**
Adds performance tests for the overview page when varying amounts of namespaces are present to understand the impact of merely having a larger number of namespaces.

Related to #5144

### CODE REVIEW

**Review by josunect — COMMENTED** [review]
It worked well in my local environment, but just with 50 namespaces. 

- I think it would be good to have a cleanup function, to just remove the namespaces, without having to remove the whole cluster. Because, when I run it with 300, my cluster was not responding well in the GUI. 
- We could include a parameter to just limit the max number of namespaces (e.x. MAX_NS = 50) 

I know the intention is not use it in a local cluster, but it could be useful for local tests or development.

File: frontend/cypress/perf/Graphload.spec.ts [hunk 1]
```diff
@@ -1,71 +1,171 @@
...
+  }
+}
+
+function deleteNamespace(name: string) {
+  cy.exec(`kubectl delete --ignore-not-found=true --wait=false ns ${name}`);
+}
+
+describe('Performance tests', () => {
+  const reportFilePath = 'logs/performance.txt';
+
```
Comment (josunect) [inline frontend/cypress/perf/Graphload.spec.ts:35]: `const maxNamespaces = (typeof Cypress.env('MAX_NS') !== 'undefined') ? Cypress.env('MAX_NS') : 300;`

File: frontend/cypress/perf/Graphload.spec.ts [hunk 2]
```diff
@@ -1,71 +1,171 @@
...
+      {
+        namespaces: 10
+      },
+      {
+        namespaces: 50
+      },
+      {
+        namespaces: 300
+      }
+    ];
```
Comment (josunect) [inline frontend/cypress/perf/Graphload.spec.ts:60]:
```
const limitedCases = [];
    overviewCases.forEach(function(testCase) {
        if (testCase.namespaces <= maxNamespaces) {
            limitedCases.push(testCase);
        }
 });
```

File: frontend/cypress/perf/Graphload.spec.ts [hunk 3]
```diff
@@ -1,71 +1,171 @@
...
-                    +"&graphType="+data.graphType
-                    +"&namespaces="+data.namespaces
-                    +"&duration="+data.duration
-                    +"&refresh="+data.refresh
-                    +"&layout="+data.layout
-                    +"&namespaceLayout="+data.namespaceLayout);
+      cy.writeFile(reportFilePath, '[Empty Namespaces]\n\n', { flag: 'a+' });
+    });
+
+    overviewCases.forEach(function (testCase) {
```
Comment (josunect) [inline frontend/cypress/perf/Graphload.spec.ts:65]: `limitedCases.forEach(function (testCase) {`

File: frontend/cypress/README.md [hunk 4]
```diff
@@ -101,6 +101,22 @@ You can use the [perf hack script](../../hack/perf-ibmcloud-openshift.sh) to spi
 make -e CYPRESS_BASE_URL="https://<kiali-openshift-route>" -e CYPRESS_PASSWD="<IBMCloud API Key>" -e CYPRESS_USERNAME="IAM#<SSO-EMAIL>" -e CYPRESS_AUTH_PROVIDER="ibmcloud" perf-tests-gui
 ```
```
Comment (josunect) [inline frontend/cypress/README.md:102]: CYPRESS_MAX_NS=10

**Comment by josunect** [comment]
> 

Yes, both comments sounds good to me, update the README file and a fixture file with the number of namespaces.

**Review by josunect — APPROVED** [review]
Changes worked fine, also, all namespaces have been cleaned up. 
The code LGTM.

**Comment by nrfox** [comment]
@leandroberetta adding you as a reviewer since you have write access and have worked with cypress before :)

**Pull request merged**